src/traces/candlestick/attributes.js

```diff
@@ -0,0 +1,49 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Lib = require('../../lib');
+var OHLCattrs = require('../ohlc/attributes');
+var boxAttrs = require('../box/attributes');
+
+var directionAttrs = {
+    visible: {
```

Comment:
and by dropping `increasing.visible` and `decreasing.visible`, it would make the `name` problem easier to solve.

For example, with

```js
trace = {
  type: 'ohlc',
  name: 'company A',
  // ...
};

// would yield

fullTrace[0] = {
  type: 'scatter',
  mode: 'lines',
  name: 'Company A - increasing',
  // ...
};

fullTrace[1] = {
  type: 'scatter',
  mode: 'lines',
  name: 'Company A - decreasing'
  // ...
};
```